src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+class FigureSupplier {
+    private static final Random random = new Random();
+    private static final double MAX_SIZE = 100.0;
+
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Color color = colorSupplier.getRandomColor();
+
+        switch (random.nextInt(5)) {
```

Comment:
Instead of using `random.nextInt(5)`, consider using a constant for the number of figure types to avoid magic numbers.